### ISSUE TITLE
New Rule: no-generic-identifier-names

### ISSUE THREAD

**Issue by pertrai1** [pertrai1/eslint-plugin-llm-core#42]
## Research Evidence

From vibecodingconsultant.com LLM patterns analysis: LLMs use vague, generic names like `data`, `result`, `value`, `item`, `tmp`.

## Description

Disallow overly generic variable and function names that don't convey meaning. LLMs frequently generate code with uninformative identifiers.

## Default Blocklist
`data`, `result`, `value`, `item`, `tmp`, `temp`, `obj`, `arr`, `val`, `res`, `ret`, `x`, `y` (in non-math contexts)

### Bad
```ts
const data = await fetchData();
const result = processData(data);
function doSomething(value) { ... }
const item = items.find(x => x.id === id);
```

### Good
```ts
const users = await fetchUsers();
const validatedUsers = validateUsers(users);
function processOrder(order: Order) { ... }
const matchingOrder = orders.find(order => order.id === id);
```

## Options
```ts
{
  "llm-core/no-generic-identifier-names": ["error", {
    "blocklist": ["data", "result", "value", "item"],
    "allowInDestructuring": true,
    "allowInTests": true
  }]
}
```

## Priority
Medium - improves code readability significantly
